### PR TITLE
Relax validation only allowing CalicoNetwork with Calico CNI plugin

### PR DESCRIPTION
**Pull request by tmjd**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/installation/validation_test.go [hunk 1]
```diff
@@ -161,20 +161,20 @@ var _ = Describe("Installation validation tests", func() {
...
-			Entry("should disallow AzureVNET", operator.PluginAzureVNET),
+			Entry("GKE", operator.PluginGKE),
+			Entry("AmazonVPC", operator.PluginAmazonVPC),
+			Entry("AzureVNET", operator.PluginAzureVNET),
 		)
 	})
 	Describe("validate non-calico CNI plugin Type", func() {
 		BeforeEach(func() {
-			instance.Spec.CalicoNetwork = nil
+			//instance.Spec.CalicoNetwork = nil
```
Comment (caseydavenport) [inline pkg/controller/installation/validation_test.go:177]: Is this intentional?

File: pkg/controller/installation/validation.go [hunk 2]
```diff
@@ -155,8 +167,14 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 					v6pool.NATOutgoing, strings.Join(operatorv1.NATOutgoingTypesString, ","))
 			}
 
-			if v6pool.NodeSelector == "" {
-				return fmt.Errorf("ipPool.nodeSelector should not be empty")
+			if instance.Spec.CNI.Type == operatorv1.PluginCalico {
+				if v6pool.NodeSelector == "" {
```
Comment (caseydavenport) [inline pkg/controller/installation/validation.go:171]: This validation isn't Calico specific, right? The node-selector must _always_ be non empty?

Comment (tmjd) [reply pkg/controller/installation/validation.go:171]: You're right.

Comment (tmjd) [reply pkg/controller/installation/validation.go:171]: I'm going to change the IPv4 code also. In both cases I'll move the `NodeSelector == ""` check to after Plugin check block. So in the case of non-Calico we'll check the NodeSelector twice.

**Pull request merged**